localstack-core/localstack/services/cloudformation/engine/validations.py

```diff
@@ -0,0 +1,81 @@
+"""
```

Comment:
Super useful. Guess we could look into making use of jsonschema for this or did you encounter any particular validations where this wouldn't work?

Reply:
That's a nice suggestion. I didn't look into it, but my concern is that we would lose the granularity of the error messages presented to the user. Perhaps we could somehow instrument the validation so we could customise the error messages, but with the framework I set out for validations I hope we could extend them further, leading on to full `jsonschema` validation in the future 👍